railties/lib/rails/generators.rb

```diff
@@ -239,8 +241,15 @@ def sorted_groups
       # Rails looks for is the first and last parts of the namespace.
       def find_by_namespace(name, base = nil, context = nil) #:nodoc:
         lookups = []
-        lookups << "#{base}:#{name}"    if base
-        lookups << "#{name}:#{context}" if context
+
+        if base
+          lookups << "#{base}:#{name}"
+          lookups << "rails:#{base}:#{name}"
+        end
+
+        if context
+          lookups << "#{name}:#{context}"
+        end
```

Comment:
What do these new lines do? I've always found the `base` and `context` naming here confusing?

Reply:
So it's possible for users to call rails g db:system:change? I thought this generator was internal and sufficiently called via it's class start method from the command?

Reply:
I'll go ahead and revert the changes to the generator lookup and hide this generator so it can only be called through the command. I'm assuming we would prefer only one way of changing databases.